app/src/main/java/com/example/topmovies/viewmodel/MovieViewModel.kt

```diff
@@ -64,30 +63,25 @@ class MovieViewModel constructor(
             EnumScreen.FAVORITE ->
                 removeMovieFromFavorites(id)
         }
-        favoriteMovies.update()
     }
 
     private fun addMovieToFavorites(id: String) {
-        val mutableMovies = movies.value!!.toMutableList()
-        val movie = movies.value!!.find { movie -> movie.id.contains(id) }
-        val copy = movie?.copy(isFavorite = true)
-        if (copy != null) {
-            mutableMovies[movies.value!!.indexOf(movie)] = copy
-            movies.postValue(mutableMovies)
-            _favoriteMovies.add(copy)
-            favoritePref.edit().putString(copy.id, "").apply()
+        repository.updateMovie(id, true)
+        repository.getMovies {
+            handler.post { movies.value = it }
+        }
+        repository.getFavoriteMovie {
+            handler.post { favoriteMovies.value = it }
```

Comment:
Are you aware that you might run into an inconsistent data scenario?
What will happen if `getFavouriteMovies()` will execute prior to `updateMovies()`?